### PR TITLE
Fix Table bug with accessor that has a period

### PR DESCRIPTION
**Pull request by stepzhou**
## Description

Same issue as https://github.com/airplanedev/web/pull/2329 but in Views.

Core issue is that React Table treats periods as object indexing.

Fixes PDS-2402

## Test plan

Write failing test and make it pass

### CODE REVIEW

File: lib/src/components/table/TableComponent.tsx [hunk 1]
```diff
@@ -731,6 +728,26 @@ const RowActionsMenu = <TRowData extends object>({
...
 
+const didColumnsChange = <TRowData extends object>(
+  currColumns: ReactTableColumn<TRowData>[],
+  newColumns: ReactTableColumn<TRowData>[]
+) => {
+  if (currColumns.length !== newColumns.length) {
+    return true;
+  }
+  for (let i = 0; i < currColumns.length; i++) {
+    const currColumnCompare = { ...currColumns[i] };
```
Comment (geodn) [inline lib/src/components/table/TableComponent.tsx:739]: `currColumn`, to be consistent with `newColumn`

File: lib/src/components/table/TableComponent.tsx [hunk 2]
```diff
@@ -731,6 +728,26 @@ const RowActionsMenu = <TRowData extends object>({
...
+    const newColumn = { ...newColumns[i] };
+    // Don't compare the accessor function since that will always be different.
+    delete currColumnCompare.accessor;
```
Comment (geodn) [inline lib/src/components/table/TableComponent.tsx:742]: It feels cleaner to me to just do something like

`const currColumn = { ...currColumns[i], accessor: undefined }`

And get rid of these delete statements

**Review by geodn — APPROVED** [review]
Couldn't an argument be made for just not allowing accessors with "."? Users are setting the accessor here, as opposed to the accessor being generated somewhat automatically. Interested to hear your take on the benefit vs messiness tradeoff. Approving to unblock since I'm fine with the change overall

**Comment by stepzhou** [comment]
> Users are setting the accessor here, as opposed to the accessor being generated somewhat automatically

But it's also tied to what the task returns. So if a task returns keys that have periods in them and the view developer doesn't own that task, I think it'd be annoying to say that you have to get the task rv changed or do a `.map` over the task output to rename the keys

**Pull request merged**